### PR TITLE
HL-673

### PR DESCRIPTION
**Pull request by bitcoinbrisbane**
Allow owners to remove vaults and markets

### CODE REVIEW

File: contracts/IVault.sol [hunk 1]
```diff
@@ -5,9 +5,11 @@ import "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
 import "@openzeppelin/contracts/interfaces/IERC4626.sol";
 
 interface IVault is IERC20Metadata, IERC4626 {
-    function getPerformance() external view returns (uint256);
+    
     function getMarket() external view returns (address);
     function getMarketAllowance() external view returns (uint256);
+    function getOwner() external view returns (address);
+    function getPerformance() external view returns (uint256);
```
Comment (bazmatic) [inline contracts/IVault.sol:12]: Actually, I thought we'd dropped `getPerformance()` given that anyone can calculate that?

File: contracts/Market.sol [hunk 2]
```diff
@@ -76,14 +76,17 @@ contract Market is IMarket, Ownable, ERC721 {
...
 		_signers[owner()] = true;
-
 		timeout = timeoutDays * 1 days;
 	}
 
 	function tokenURI(uint256 tokenId) public view virtual override returns (string memory) {
         return string(abi.encodePacked(baseURI, Strings.toHexString(uint256(uint160(_self)), 20), "/", tokenId.toString()));
     }
 
+	function getOwner() external view returns (address) {
```
Comment (bazmatic) [inline contracts/Market.sol:86]: This is redundant as `owner()` is already a public function in `Ownable`

```    function owner() public view virtual returns (address) {
        return _owner;
    }
```

File: tests/registry.test.ts [hunk 3]
```diff
@@ -112,4 +114,101 @@ describe("Registry", () => {
 		const vault_count2 = await registry.vaultCount();
 		expect(vault_count2, "Should have 1 vault").to.equal(1);
 	});
+
+	it.only("Should be able to remove a market", async () => {
```
Comment (bazmatic) [inline tests/registry.test.ts:118]: Oop

Comment (bazmatic) [inline tests/registry.test.ts:118]:
```suggestion
	it("Should be able to remove a market", async () => {
```

File: tests/registry.test.ts [hunk 4]
```diff
@@ -112,4 +114,101 @@ describe("Registry", () => {
...
+		await registry.addMarket(mockMarket2.address);
+		await registry.addMarket(mockMarket3.address);
+
+		let market_count = await registry.marketCount();
+		expect(market_count, "Should have 3 markets").to.equal(3);
+
+		await registry.removeMarket(1, mockMarket2.address);
+
+		market_count = await registry.marketCount();
+		expect(market_count, "Should have 2 markets").to.equal(2);
```
Comment (bazmatic) [inline tests/registry.test.ts:172]: It would be good if the test demonstrated that the correct market was removed.

File: tests/registry.test.ts [hunk 5]
```diff
@@ -112,4 +114,101 @@ describe("Registry", () => {
...
+		await registry.addVault(mockVault2.address);
+		await registry.addVault(mockVault3.address);
+
+		let vault_count = await registry.vaultCount();
+		expect(vault_count, "Should have 3 vaults").to.equal(3);
+
+		await registry.removeVault(1, vault.address);
+
+		vault_count = await registry.vaultCount();
+		expect(vault_count, "Should have 2 vaults").to.equal(2);
```
Comment (bazmatic) [inline tests/registry.test.ts:212]: Ditto. 2 left, but is it the correct 2?

**Review by bazmatic — APPROVED** [review]

File: contracts/Market.sol (hunk 2 above)
Comment (brettvniekerk) [reply contracts/Market.sol:86]: +1, just call `owner()`

Comment (bitcoinbrisbane) [reply contracts/Market.sol:86]: Owner is not on the IVault interface.  Needs to be wrapped.

**Pull request merged**